tests/Unit/Evolution/Systems/ScalarTensor/BoundaryConditions/Test_ProductOfConditions.cpp

```diff
@@ -0,0 +1,377 @@
+  std::optional<std::string> scalar_result_ghost;
+  std::optional<std::string> product_result_ghost;
+
+  std::optional<std::string> gh_result_time_derivative;
+  std::optional<std::string> scalar_result_time_derivative;
+  std::optional<std::string> product_result_time_derivative;
+
+  if constexpr (DerivedGhCondition::bc_type ==
+                evolution::BoundaryConditions::Type::Ghost) {
+    gh_result_ghost = gh_bc_helper::dg_ghost(
+        make_not_null(&expected_mutable_variables), argument_variables,
+        gridless_box, derived_gh_condition);
+  }
+
+  if constexpr (DerivedScalarCondition::bc_type ==
+                evolution::BoundaryConditions::Type::Ghost) {
+    scalar_result_ghost = scalar_bc_helper::dg_ghost(
+        make_not_null(&expected_mutable_variables), argument_variables,
+        gridless_box, derived_scalar_condition);
+  }
+
+  if constexpr (bc_type == evolution::BoundaryConditions::Type::Ghost) {
+    product_result_ghost = product_bc_helper::dg_ghost(
+        make_not_null(&mutable_variables), argument_variables, gridless_box,
+        derived_product_condition);
+  }
+
+  if (gh_result_ghost.has_value()) {
+    if (scalar_result_ghost.has_value()) {
+      CHECK((product_result_ghost.value() ==
+             gh_result_ghost.value() + ";" + scalar_result_ghost.value()));
+    } else {
+      CHECK(product_result_ghost == gh_result_ghost);
+    }
+  } else {
+    CHECK(product_result_ghost == scalar_result_ghost);
+  }
+
+  CHECK_VARIABLES_APPROX(mutable_variables, expected_mutable_variables);
+}
+}  // namespace
+
+SPECTRE_TEST_CASE(
+   "Unit.Evolution.Systems.ScalarTensor.BoundaryConditions.ProductOfConditions",
+   "[Unit][Evolution]") {
+  register_factory_classes_with_charm<Metavariables>();
+  // scoped to separate out each product combination
+  {
+    INFO(
+        "Product condition of DirichletAnalytic for Generalized Harmonic and "
+        "AnalyticConstant for CurvedScalarWave.");
+
+    const gh::BoundaryConditions::DirichletAnalytic<3_st> gh_condition{
+        std::unique_ptr<evolution::initial_data::InitialData>(
+            std::make_unique<gh::Solutions::WrappedGr<
+                ScalarTensor::AnalyticData::KerrSphericalHarmonic>>(
+                // Black Hole parameters
+                1.0, std::array<double, 3>{{0.1, -0.2, 0.3}},
+                // Scalar wave parameters
+                2.0, 1.0, 1.0, std::pair<size_t, int>{1, 0}))};
+    const CurvedScalarWave::BoundaryConditions::AnalyticConstant<3_st>
+        scalar_condition{// Amplitude
+                         0.0};
+    const auto product_boundary_condition = TestHelpers::test_creation<
+        std::unique_ptr<ScalarTensor::BoundaryConditions::BoundaryCondition>,
+        Metavariables>(
+        "ProductDirichletAnalyticAndAnalyticConstant:\n"
+        "  GeneralizedHarmonicDirichletAnalytic:\n"
+        "    AnalyticPrescription:\n"
+        "      KerrSphericalHarmonic:\n"
+        "        Mass: 1.0\n"
```

Comment:
Generally using non-unity numbers makes the test more reliable. Would that be possible for the radius & width too?

Reply:
I've changed all test parameters to be non-zero/unity and non-integer.